### PR TITLE
SDIT-513: ⬆️  Upgrade to latest spring boot

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/config/ResourceServerConfiguration.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ import org.springframework.security.web.server.SecurityWebFilterChain
 
 @Configuration
 @EnableWebFluxSecurity
-@EnableReactiveMethodSecurity
+@EnableReactiveMethodSecurity(useAuthorizationManager = false)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/config/ResourceServerConfiguration.kt:13]: bug fix after latest spring boot upgrade

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/data/CreateRoomMappingDto.kt [hunk 2]
```diff
@@ -13,6 +13,6 @@ data class CreateRoomMappingDto(
   @Schema(description = "nomis room description (unique within prison)", required = true)
   val nomisRoomDescription: String,
 
-  @Schema(description = "open or closed indicator", required = true)
-  val isOpen: Boolean,
+  @Schema(description = "open or closed indicator, default is false")
+  val isOpen: Boolean = false,
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/data/CreateRoomMappingDto.kt:17]: Not expecting any functional changes - though I guess this is non-breaking.

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/data/CreateRoomMappingDto.kt:17]: agreed, wasn't expecting to see that either.  I'll check to see what has gone wrong

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt [hunk 3]
```diff
@@ -789,21 +784,23 @@ class IncentiveMappingResourceIntTest : IntegrationTestBase() {
         .jsonPath("totalElements").isEqualTo(4)
         .jsonPath("$.content..nomisBookingId").value(
           Matchers.contains(
-            1, 2, 3, 4
-          )
+            1,
+            2,
+            3,
+            4,
+          ),
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt:791]: Could do better ktlint (such as joining these into a single line).

Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt:791]: Could do better ktlint (such as joining these into a single line).

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/data/CreateRoomMappingDto.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/data/CreateRoomMappingDto.kt:17]: Ah it's opk - there was a fix from Lucy in the middle of the commits

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**